backend/danswer/chat/process_message.py

```diff
@@ -446,6 +486,13 @@ def stream_chat_message_objects(
                     tool_dict[db_tool_model.id] = [
                         ImageGenerationTool(api_key=dalle_key)
                     ]
+                elif tool_cls.__name__ == InternetSearchTool.__name__:
+                    bing_api_key = BING_API_KEY
```

Comment:
Totally not needed but I wonder if it makes sense to have an internet search tool interface and have bing be an implementation of it. That way the bing specific implementation details isn't nested deeply into other flows

Reply:
I did this in my other search project, but I made the interface once someone specifically requested another search provider. What do you think about this?